lib/pipeline.js

```diff
@@ -849,7 +849,7 @@ class PipelineModel extends BaseModel {
         // create or update child pipelines
         newScmUrls.forEach(scmUrl => toCreateOrUpdate.push(this._createOrUpdatePipeline(scmUrl)));
 
-        return Promise.resolve(toCreateOrUpdate);
+        return Promise.all(toCreateOrUpdate);
```

Comment:
now since we are really waiting, we would need error handling for each of the promise, it shouldn't break in there is issue with an scmURL